### PR TITLE
support java.util.List

### PR DESCRIPTION
**Pull request by tboonIV**
to address #12

### CODE REVIEW

File: examples/src/main/java/io.github.lwlee2608.proto.example/helloworld/HelloRequest.java [hunk 1]
```diff
@@ -5,10 +5,13 @@
...
 
+import java.util.List;
+
 @Data
 @Accessors(chain = true)
 @ProtoMessage(protoName = "helloworld", protoPackage = "example.helloworld")
 public class HelloRequest {
     @ProtoField(tag = 1) private String message;
     @ProtoField(tag = 2) private Integer id;
+    @ProtoField(tag = 3) private List<String> names;
```
Comment (lwlee2608) [inline examples/src/main/java/io.github.lwlee2608.proto.example/helloworld/HelloRequest.java]: Add a list field to `AllTypePayload` instead

File: proto-java-annotation/src/main/java/io/github/lwlee2608/proto/annotation/processor/Field.java [hunk 2]
```diff
@@ -12,5 +12,6 @@ public class Field {
     private Integer tag;
     private Boolean isStruct;
     private Boolean isEnum;
+    private Boolean isList;
     // private Boolean isList; // TODO
```
Comment (lwlee2608) [inline proto-java-annotation/src/main/java/io/github/lwlee2608/proto/annotation/processor/Field.java]: Remove the duplicate comment

File: proto-java-annotation/src/main/java/io/github/lwlee2608/proto/annotation/processor/ProtoProcessor.java [hunk 3]
```diff
@@ -136,6 +137,10 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
                             protoType = getSimpleClass(javaType) + "Enum";
                             isEnum = true;
 
+                        } else if (javaType.startsWith("java.util.List")) {
+                            String subJavaType = javaType.substring(javaType.indexOf("<")+1, javaType.indexOf(">"));
+                            protoType = "repeated " + toProtoType(subJavaType);
```
Comment (lwlee2608) [inline proto-java-annotation/src/main/java/io/github/lwlee2608/proto/annotation/processor/ProtoProcessor.java]: `protoType` should just be the raw data type.

**Review by lwlee2608 — CHANGES_REQUESTED** [review]
Also we need to populate a list field in `testHelloWorldWithPayload` unit test

File: proto-java-annotation/src/main/java/io/github/lwlee2608/proto/annotation/processor/ProtoProcessor.java (hunk 3 above)
Comment (tboonIV) [reply proto-java-annotation/src/main/java/io/github/lwlee2608/proto/annotation/processor/ProtoProcessor.java]: so we don't support null value in list<String> ?

Comment (lwlee2608) [reply proto-java-annotation/src/main/java/io/github/lwlee2608/proto/annotation/processor/ProtoProcessor.java]: a `protoType` should be just `int32`. `string` and so on. `repeated int32` is not a protoType.
You should print `repeated` in `writeProtoFile()` function

**Review by lwlee2608 — APPROVED** [review]
Looks good, thank you for your contribution.

**Pull request merged**